Bugsnag/Breadcrumbs/BugsnagBreadcrumbs.m

```diff
@@ -137,52 +137,56 @@ - (void)setCapacity:(NSUInteger)capacity {
 }
 
 - (NSUInteger)count {
-    return self.breadcrumbs.count;
+    __block NSUInteger count;
+    dispatch_barrier_sync(self.readWriteQueue, ^{
+        count = self.breadcrumbs.count;
+    });
+    return count;
 }
 
 - (BugsnagBreadcrumb *)objectAtIndexedSubscript:(NSUInteger)index {
-    if (index < [self count]) {
-        __block BugsnagBreadcrumb *crumb = nil;
-        dispatch_barrier_sync(self.readWriteQueue, ^{
-          crumb = self.breadcrumbs[index];
-        });
-        return crumb;
-    }
-    return nil;
+    __block BugsnagBreadcrumb *crumb = nil;
+    dispatch_barrier_sync(self.readWriteQueue, ^{
+        if (index < self.breadcrumbs.count) {
+            crumb = self.breadcrumbs[index];
+        }
+    });
+    return crumb;
 }
 
 - (NSArray *)arrayValue {
-    if ([self count] == 0) {
-        return nil;
-    }
-    __block NSMutableArray *contents =
-        [[NSMutableArray alloc] initWithCapacity:[self count]];
+    __block NSMutableArray *contents = nil;
     dispatch_barrier_sync(self.readWriteQueue, ^{
-      for (BugsnagBreadcrumb *crumb in self.breadcrumbs) {
-          NSDictionary *objectValue = [crumb objectValue];
-          NSError *error = nil;
-          @try {
-              if (![NSJSONSerialization isValidJSONObject:objectValue]) {
-                  bsg_log_err(@"Unable to serialize breadcrumb: Not a valid "
-                              @"JSON object");
-                  continue;
-              }
-              [contents addObject:objectValue];
-          } @catch (NSException *exception) {
-              bsg_log_err(@"Unable to serialize breadcrumb: %@", error);
-          }
-      }
+        NSUInteger numBreadcrumbs = self.breadcrumbs.count;
+        if (numBreadcrumbs > 0) {
```

Comment:
nit: removing this check would mean that the method never needs to return `nil` and could be annotated as nonnull - which feels a better behaviour for an empty collection.